### PR TITLE
dialog handler unit test

### PR DESCRIPTION
**Pull request by vrbsm**
## Description
dialog handler unit test

## Solution Approach
N/A

## Documentation
N/A

## Related PRs
N/A

## QA Checklist
- [x] Unit-Tests
- [ ] Automation-Tests

Attach unit & spectron tests results in PDF format against the above task lists for this branch
Screen Shot 2019-03-28 at 15 07 59

### CODE REVIEW

File: spec/dialogHandler.spec.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            const requestMocked = {
+                url: 'https://symphony.corporate.com/',
+            };
+            const authInfoMocked = {
+                host: 'symphony.com',
+            };
+            ipcRenderer.send('login', webContentsMocked, requestMocked, authInfoMocked, callbackMocked);
+            expect(spy).toBeCalledWith(webContentsMocked, 'symphony.com', true, expect.any(Function), callbackMocked);
+        });
+        describe('certificate-error', () => {
```
Comment (KiranNiranjan) [inline spec/dialogHandler.spec.ts]: leave a space here

File: spec/dialogHandler.spec.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            const urlMocked = 'https://symphony.corporate.com/';
+            const errorMocked = 'check for server certificate revocation';
+            const certificate = null;
+            it('should return false when buttonId is equal 1', () => {
+                dialog.showMessageBox = jest.fn(() => 1);
+                ipcRenderer.send('certificate-error', webContentsMocked, urlMocked, errorMocked, certificate, callbackMocked);
+                expect(callbackMocked).toBeCalledWith(false);
+            });
+            it('should return true when buttonId is not equal 1', () => {
```
Comment (KiranNiranjan) [inline spec/dialogHandler.spec.ts]: same here

File: spec/dialogHandler.spec.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                dialog.showMessageBox = jest.fn(() => 2);
+                ipcRenderer.send('certificate-error', webContentsMocked, urlMocked, errorMocked, certificate, callbackMocked);
+                expect(callbackMocked).toBeCalledWith(true);
+                ipcRenderer.send('certificate-error', webContentsMocked, urlMocked, errorMocked, certificate, callbackMocked);
+                expect(callbackMocked).toBeCalledWith(true);
+
```
Comment (KiranNiranjan) [inline spec/dialogHandler.spec.ts]: remove extra space

File: spec/dialogHandler.spec.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+describe('dialog handler', () => {
+    const callbackMocked = jest.fn();
+    const webContentsMocked = jest.fn();
+
+    beforeEach(() => {
+        jest.clearAllMocks().resetModules();
+    });
+
+    describe('events', () => {
+        it('login', () => {
```
Comment (VishwasShashidhar) [inline spec/dialogHandler.spec.ts]: please change the name from `login` to `should login correctly`

File: spec/dialogHandler.spec.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                host: 'symphony.com',
+            };
+            ipcRenderer.send('login', webContentsMocked, requestMocked, authInfoMocked, callbackMocked);
+            expect(spy).toBeCalledWith(webContentsMocked, 'symphony.com', true, expect.any(Function), callbackMocked);
+        });
+        describe('certificate-error', () => {
+            const urlMocked = 'https://symphony.corporate.com/';
+            const errorMocked = 'check for server certificate revocation';
+            const certificate = null;
+            it('should return false when buttonId is equal 1', () => {
```
Comment (VishwasShashidhar) [inline spec/dialogHandler.spec.ts]: change the name to `should return false when buttonId is 1`

File: spec/dialogHandler.spec.ts (hunk 2 above)
Comment (VishwasShashidhar) [inline spec/dialogHandler.spec.ts]: change the name to `should return true when buttonId is not 1`

**Pull request merged**